restli-tools/src/main/resources/apiVmTemplates/action.vm

```diff
@@ -0,0 +1,129 @@
+#*
+Copyright (c) 2021 LinkedIn Corp.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*#
+
+  #foreach($method in $spec.actions)
+    #set($actionParamClassName = "${util.nameCapsCase($method.name)}ActionParameters")
+    #set($actionOptionalParamClassName = "${util.nameCapsCase($method.name)}ActionOptionalParameters")
+    #doc($method.schema.doc)
+
+    #if(${method.hasRequiredParams()}) ## action with no required Params will only have one API
+    @SuppressWarnings("unchecked")
+    public CompletionStage<${method.valueClassDisplayName}> ${util.nameCamelCase(${method.name})}(
+      #if(${method.isEntityAction()})
+      $spec.keyClassName $spec.idName,
+      #end
+        #set($paramLen = ${method.getRequiredParameters().size()})
+        #if($paramLen > 0)
+          #set($lastParam = ${method.getRequiredParameters().get($paramLen - 1)})
+          #if($paramLen > 1)
+            #foreach($idx in [2..$paramLen])
+              #set($param = ${method.getRequiredParameters().get($idx - 2)})
```

Comment:
Can you do `ids in [0..$paramLen-2]` instead?

Reply:
this is what I wrote initially. But the velocity template doesn't support this syntax(I will have to define an additional variable to make it work).